lua/orgmode/org/links/handlers/_internal/custom_id.lua

```diff
@@ -0,0 +1,53 @@
+local utils = require('orgmode.utils')
+local Org = require('orgmode')
+local Internal = require('orgmode.org.hyperlinks.builtin.internal')
+
+---@class OrgLinkHandlerCustomId:OrgLinkHandlerInternal
+local CustomId = Internal:new()
+
+function CustomId:new(custom_id)
+  ---@class OrgLinkHandlerCustomId
+  local this = Internal:new()
+  this.custom_id = custom_id
+  setmetatable(this, self)
+  self.__index = self
+  return this
+end
+
+---@param input string
+function CustomId.parse(input)
+  return CustomId:new(input)
+end
+
+function CustomId:__tostring()
+  return string.format('#%s', self.custom_id)
+end
+
+function CustomId:follow()
+  local headlines = Org.files:get_current_file():find_headlines_with_property_matching('custom_id', self.custom_id)
+
+  if #headlines == 0 then
+    return utils.echo_warning(('Could not find custom ID "%s".'):format(self.custom_id))
+  end
+
+  self.goto_oneof(headlines)
+end
+
+function CustomId:insert_description()
+  return self.custom_id
+end
+
+function CustomId:complete(lead, context)
+  local file = self.get_file_from_context(context)
+  local headlines = file:find_headlines_with_property_matching('CUSTOM_ID', lead)
+
+  local completions = {}
+  for _, headline in pairs(headlines) do
+    local id = headline:get_property('CUSTOM_ID')
+    table.insert(completions, self:new(id):__tostring())
```

Comment:
Since we are using the metamethod to set up a string format of an entry, we should use `tostring(self:new(id))`. Other option is to have a custom `to_string` method and call that here directly. I'm ok with any of the options.

Reply:
Used `tostring`, just forgot it existed tbh.